nodejs/src/apis/RoomService.ts

```diff
@@ -1,103 +1,111 @@
-import { APIService } from "./APIService";
-import { HMSRoom } from "./interfaces/roomInterfaces";
+import { logger } from "../services/LoggerService";
+import { APIService } from "../services/APIService";
 import { QueryResults } from "./interfaces/common";
-import { logger } from "../LoggerService";
+import {
+  HMSCreateRoomConfig,
+  HMSRoom,
+  HMSUpdateRoomConfig as HMSUpdateRoomOptions,
+} from "./interfaces/roomInterfaces";
+import { QueryResultsIterator } from "../utils/QueryResultsIterator";
 
 export class RoomService {
   private basePath = "/rooms";
 
   constructor(private apiService: APIService) {}
 
-  async getRoomById(roomId: string): Promise<HMSRoom> {
-    return this.apiService.get(`${this.basePath}/${roomId}`);
+  /**
+   *
+   * @returns a `QueryResultsIterator<HMSRoom>` object
+   */
+  getRoomsIterable(): QueryResultsIterator<HMSRoom> {
+    const queryResultsIterable = new QueryResultsIterator<HMSRoom>(
+      (queryParams: Record<string, any>) => this.apiService.get(this.basePath, queryParams),
+      {}
+    );
+    return queryResultsIterable;
   }
 
-  async getHlsStateByRoomId(roomId: string): Promise<HLSRoomState> {
-    const data: Record<string, any> = await this.apiService.get("/meetings/hls", {
-      room_id: roomId,
-    });
-    return {
-      recording: {
-        hlsVod: data.recording?.hls_vod,
-        singleFilePerLayer: data.recording?.single_file_per_layer,
-      },
-      running: data.running,
-      url: data.url,
-    };
+  /**
+   *
+   * @param roomId Room ID
+   * @returns a `HMSRoom` object
+   */
+  async getRoomById(roomId: string): Promise<HMSRoom> {
+    return this.apiService.get(`${this.basePath}/${roomId}`);
   }
 
+  /**
+   *
+   * @param name Room name
+   * @returns a `HMSRoom` object
+   */
   async getRoomByName(name: string): Promise<HMSRoom> {
     const results: QueryResults<HMSRoom> = await this.apiService.get(this.basePath, { name });
-    if (results.data.length === 0) {
-      const err = new Error(`no room found with passed in name - ${name}`);
-      logger.error("no room found", err);
+    if (!results.data || results.data.length === 0) {
+      const err = new Error(`no Roomfound with passed in name - ${name}`);
+      logger.error("no Roomfound", err);
       throw err;
     }
     return results.data[0];
   }
 
-  async getHlsURL(roomId: string): Promise<string> {
-    const result: HLSRoomState = await this.apiService.get("/meetings/hls/url", {
-      room_id: roomId,
-    });
-    return result.url || "";
+  /**
+   *
+   * @param enabled Enabled status of Room
+   * @returns a `QueryResultsIterator<HMSRoom>` object
+   */
+  getRoomsByEnabledIterable(enabled: boolean): QueryResultsIterator<HMSRoom> {
```

Comment:
should we not instead have `getRoomsByFiltersIterable(listRoomFilters: {..})`?

what if for e.g. I want all enabled rooms within some time range?